src/gui/src/layoutViewer.cpp

```diff
@@ -224,9 +227,16 @@ class GuiPainter : public Painter
     painter_->drawLine(x - o, y + o, x + o, y - o);
   }
 
-  const odb::Point determineStringOrigin(int x, int y, Anchor anchor, const QString& text, bool rotate_90 = false)
+  const odb::Point determineStringOrigin(int x, int y, Anchor anchor, const QString& text, bool rotate_90 = false, const QRect *bounding_rect = nullptr)
```

Comment:
Since we use the boundingRect (alot) maybe it would be possible to cache it centrally, so we can look up the bounding box if its computed? This is just one of many places we use this and if this adds a lot of time, there would be a lot of value in correcting that.

Reply:
We should be able to  create a QCache of them. I'll add this.